AUTH.md

```diff
@@ -0,0 +1,304 @@
+"delete" permission. The current entities that exist are:
+
+- job
+- garden
+- queue
+- request
+- system
+
+Permissions are defined as strings of the format "\<entity\>:\<operation\>". For
+example: `garden:read`, `system:update`, `request:delete`.
+
+There are also a limited number of special permissions that do not map to a
+typical entity or operation. These currently include:
+
+- `event:forward` - This permission is required for garden to garden
+  communications. When tasking a remote garden, the user that the local garden
+  connects to the remote garden as would need to have this permission on the
+  remote garden if authorization is also enabled on the remote garden. **NOTE:**
+  Regular users should not have this permission, as it allows for creation of
+  requests against any garden or system.
+
+## Authentication Basics
+
+When authorization is enabled on your garden, users need a way to authenticate
+in order to access the garden using their account. From a user's perspective,
+this is done via the following:
+
+### Web UI Access
+
+When authorization is enabled, a "login" button will appear toward the top right
+of the page. Clicking this brings up a login box for the user to enter their
+credentials and sign in.
+
+### API Access
+
+For direct API access, a user would first send their credentials in a POST to
+`/api/v1/token`. This results in an access and refresh token being provided back
+in the response. The access token would then be provided by the user in
+subsequent API calls via the `Authorization: Bearer` header.
+
+This token retrieval and usage is handled for the user by the Web UI and
+EasyClient, but both use this API login and access token workflow behind the
+scenes.
+
+## Auth Settings
+
+Authorization and authentication settings are housed under a top level `auth`
+section of the main application configuration yaml file. The available settings
+are:
+
+### authentication_handlers
+
+This section allows you to configure the ways that users are able to
+authenticate to the garden. The available handlers and their respective settings
+are:
+
+#### basic
+
+Allows for username and password authentication. Available settings:
+
+- **enabled:** Set to `true` to allow username and password authentication.
+
+#### trusted_header
+
+Allow for users to be authenticated via request headers that get set via a
+trusted proxy. If your garden sits behind a proxy that will authenticate the
+user and place their username in a header, enabling this will tell the garden to
+use that provided username.
+
+When logging in via the UI, the login dialog will always show the input fields
+for username and password. If a user is authenticating by providing certificates
+on all requests that go through a proxy, these fields can safely be left blank.
+Since the trusted headers will be included on the login request, the user will
+still be able to login.
+
+**NOTE:** If you enable this it is imperitive that users are required to access
```

Comment:
I think this needs a comma: *If you enable this, it is imperative...*. Also, spelling of *imperative*.

Reply:
Changed.